frontend/javascripts/oxalis/view/action-bar/starting_job_modals.tsx

```diff
@@ -675,13 +675,23 @@ function useCurrentlySelectedBoundingBox(
 ): UserBoundingBox | undefined {
   const selectedBoundingBoxId = Form.useWatch("boundingBoxId", form);
   const currentlySelectedLayerName = Form.useWatch("layerName", form);
-  const currentSelectedLayer = layers.find((layer) => layer.name === currentlySelectedLayerName);
-  const indexOfLayer = currentSelectedLayer ? layers.indexOf(currentSelectedLayer) : -1;
-  return isBoundingBoxConfigurable
-    ? userBoundingBoxes.find((bbox) => bbox.id === selectedBoundingBoxId)
-    : indexOfLayer >= 0
-      ? defaultBBForLayers[indexOfLayer]
-      : undefined;
+  const [currentlySelectedBoundingBox, setCurrentlySelectedBoundingBox] = useState<
+    UserBoundingBox | undefined
+  >(undefined);
+  // userBoundingBoxes, defaultBBForLayers, layers are different objects with each calls,
+  // but they shouldn't be able to change while the modal is open
+  // biome-ignore lint/correctness/useExhaustiveDependencies: see above
+  useEffect(() => {
+    const currentSelectedLayer = layers.find((layer) => layer.name === currentlySelectedLayerName);
+    const indexOfLayer = currentSelectedLayer ? layers.indexOf(currentSelectedLayer) : -1;
+    const newCurrentlySelectedBoundingBox = isBoundingBoxConfigurable
+      ? userBoundingBoxes.find((bbox) => bbox.id === selectedBoundingBoxId)
+      : indexOfLayer >= 0
+        ? defaultBBForLayers[indexOfLayer]
+        : undefined;
+    setCurrentlySelectedBoundingBox(newCurrentlySelectedBoundingBox);
+  }, [selectedBoundingBoxId, currentlySelectedLayerName, isBoundingBoxConfigurable]);
+  return currentlySelectedBoundingBox;
```

Comment:
This is the fix to avoid infinetly requesting the job costs. This was done due to previously the selected bounding box always being a new object and thus the cache check of whether the costs should be re-requested was always set to true. Now the selected bounding box is correctly stores as a state variable and only updated in case the user edits info in the form.

Some variables used in the effect guarding the re-calculation of the selected bounding box are not included in the dependency list, as they are also newly computed each rendering cycle and thus would always lead to the effect being recalculated

Reply:
@fm3 do you think you can also review this, or should I ping daniel?